app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,610 @@
+            default:
+                text = {
+                    id: intl.formatMessage({id: 'mobile.integration_selector.loading_options'}),
+                    defaultMessage: 'Loading Options...',
+                };
+                break;
+        }
+
+        return (
+            <View style={style.loadingContainer}>
+                <FormattedText
+                    {...text}
+                    style={style.loadingText}
+                />
+            </View>
+        );
+    }, [style, dataSource, loading, intl]);
+
+    const renderNoResults = useCallback((): JSX.Element | null => {
+        if (loading || page.current === INITIAL_PAGE) {
+            return null;
+        }
+
+        return (
+            <View style={style.noResultContainer}>
+                <FormattedText
+                    id='mobile.custom_list.no_results'
+                    defaultMessage='No Results'
+                    style={style.noResultText}
+                />
+            </View>
+        );
+    }, [loading, style]);
+
+    const renderChannelItem = (itemProps: any) => {
+        const itemSelected = Boolean(multiselectSelected[itemProps.item.id]);
+        return (
+            <ChannelListRow
+                key={itemProps.id}
+                {...itemProps}
+
+                theme={theme}
+                channel={itemProps.item as Channel}
+                selectable={isMultiselect || false}
+                selected={itemSelected}
+            />
+        );
+    };
+
+    const renderOptionItem = (itemProps: any) => {
+        const itemSelected = Boolean(multiselectSelected[itemProps.item.value]);
+        return (
+            <OptionListRow
+                key={itemProps.id}
+                {...itemProps}
+                theme={theme}
+                selectable={isMultiselect || false}
+                selected={itemSelected}
+            />
+        );
+    };
+
+    const renderUserItem = (itemProps: any): JSX.Element => {
+        const itemSelected = Boolean(multiselectSelected[itemProps.item.id]);
+
+        return (
+            <UserListRow
+                key={itemProps.id}
+                {...itemProps}
+                theme={theme}
+                selectable={isMultiselect || false}
+                user={itemProps.item}
+                teammateNameDisplay={teammateNameDisplay}
+                selected={itemSelected}
+            />
+        );
+    };
+
+    const getRenderItem = useCallback((): (itemProps: any) => JSX.Element => {
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS:
+                return renderUserItem;
+            case ViewConstants.DATA_SOURCE_CHANNELS:
+                return renderChannelItem;
+            default:
+                return renderOptionItem;
+        }
+    }, [dataSource, renderUserItem, renderChannelItem, renderOptionItem]);
+
+    const renderSelectedOptions = useCallback((): React.ReactElement<any, string> | null => {
```

Comment:
If I am not mistaken, you are not passing this function to any  prop, not using it in any dependency, just calling it a few lines later, so no need for usecallback.